src/rgw/driver/rados/rgw_rados.cc

```diff
@@ -8541,7 +8746,7 @@ int RGWRados::cls_obj_complete_op(BucketShard& bs, const rgw_obj& obj, RGWModify
   cls_rgw_obj_key key(ent.key.name, ent.key.instance);
   cls_rgw_guard_bucket_resharding(o, -ERR_BUSY_RESHARDING);
   cls_rgw_bucket_complete_op(o, op, tag, ver, key, dir_meta, remove_objs,
-                             svc.zone->need_to_log_data(), bilog_flags, &zones_trace);
+                             svc.zone->need_to_log_data(), bilog_flags, &zones_trace, obj.key.get_loc());
```

Comment:
why does the index completion need the locator here, when it didn't before?

Reply:
Previously (normally) the locator was handled during the prepare phase. But I'm skipping the prepare phase and just doing the complete, and I need to be able to set the locator.

Reply:
> note that set_olh() still calls olh_init_modification() which writes a RGWOLHPendingInfo xattr to the olh head object

As for the earlier comment, as shown below, the OLH pending ops are cleaned up.